### PR TITLE
Extra DB availability script

### PR DESCRIPTION
**Pull request by bernardmaltais**
Adding the ability to provide a custom MySQL DB Availability script. The current one in the chart is so verbose that it causes issue with looking at logs in Kibana. Keeping it much less verbose via a custom script will significantly improve log readability.

### CODE REVIEW

**Comment by zachomedia** [comment]
Thanks @bernardmaltais! I wonder if we should just always make it a value in values.yaml and set the default to current script? Thoughts @bernardmaltais  and @sylus? Then perhaps we could call the setting `dbAvailabilityScript`

**Comment by bernardmaltais** [comment]
@zachomedia I tried to make the update "transparent" to existing users. If it was implemented as you proposed would it nor result in a braking update for users as they would need to add the missing entries in the values file? I am  always for adding changes that are not braking unless it is really necessary.

**Comment by zachomedia** [comment]
> In fact many other parts of the job scripts could benefit from defaults substitution in the values file. Many times I wished I could customize the backup or restore jobs on a per env basis but the way the chart is currently written it is hard coded... so I think much more work could be done here to allow better customization across the jobs scripts for certain sections.

I think this is a great idea! It was, honestly, built initial for a specific use case but I think it can greatly benefit from being expanded and more customizable. Even other stuff like the migrations that are executed would benefit in being customizable. Default values via the values.yaml I think is the best and cleanest approach.

> Once it is decided what is best moving forward I could start to submit updates to allow customization for the different sections of the jobs.

I'll let @sylus jump in with his thoughts before we finalize on a direction for this change.

**Comment by bernardmaltais** [comment]
I converted the code to use the internal values.yaml instead. Much cleaner and should be super easy for users to customize.

**Comment by sylus** [comment]
Yup I completely agree here and think would let some of the charts end users to better override and customize their settings but showing a nice default. +1 from me :)

**Comment by bernardmaltais** [comment]
@sylus Sound good. I knew you would land on the values file so I already made the changes ;-) How would you recommend we modify the rest of the script to support values.yaml customization? Create a separate pull request with all the changes bundled into one update? Or should I just piggy back on this one to add the other ones?

**Comment by bernardmaltais** [comment]
Hummm... this could run pretty deep... I can see that allowing cistomization could have possibly a profound impact... for example:

```yaml
              echo "Restoring backup: $BACKUPNAME"

              {{- if .Values.drupal.restore.db }}
              echo "Drop database"
              drush sql-drop -y
              echo "Restore database"
              gunzip -c /backup/$BACKUPNAME/db.sql.gz{{ if .Values.drupal.restore.convert }} | sed 's/MyISAM/InnoDB/g'{{ end }} | drush sql-cli
              echo "Database restored"
              {{- end }}
```

This whole block that does the if could be totally removed since it is only used in this one script... SO instead of doing it this way it could be folded as a script customization... but this would be a breaking change...

SO I guess we need to discuss how we add customization to the different blocks... One size fits all might not work here.

**Comment by bernardmaltais** [comment]
I did a quick test and apparently helm will not render things like {{- if .Values.external.enabled }} received as part of a value... so this will limit what can be done from a script customization perspective... pretty much limiting customization to script blocks that do not include yaml syntax...

File: drupal/values.yaml [hunk 1]
```diff
@@ -111,6 +111,12 @@ drupal:
   #  drush config-set system.performance js.preprocess 0 -y;
   #  drush config-set system.performance css.preprocess 0 -y;
 
+  # -- default script used to detect when the DB is ready
+  DBAvailabilityScript:
```
Comment (zachomedia) [inline drupal/values.yaml:115]: @bernardmaltais I wonder if we should name this `dbAvailabilityScript`? All the other keys start with lowercase.

**Comment by zachomedia** [comment]
@bernardmaltais Apologies for the delay.. I think this is looking good, I have just one question that I put on the MR. Once we decide one way or another I'm happy to merge this and accept future updates :)

File: drupal/values.yaml (hunk 1 above)
Comment (bernardmaltais) [reply drupal/values.yaml:115]: I was wondering about that. Not an issue. Do you want me to make the change?

**Comment by bernardmaltais** [comment]
@zachomedia Sound good. I have updated the code with the new value name and revved up the chart version.

**Pull request merged**